lte/gateway/c/oai/tasks/amf/Registration.cpp

```diff
@@ -0,0 +1,856 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sstream>
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "log.h"
+#include "3gpp_24.501.h"
+#include "conversions.h"
+#ifdef __cplusplus
+}
+#endif
+#include "amf_data.h"
+#include "amf_fsm.h"
+#include "amf_app_ue_context_and_proc.h"
+#include "amf_asDefs.h"
+#include "amf_as.h"
+#include "amf_sap.h"
+#include "M5GSRegistrationResult.h"
+#include "amf_recv.h"
+#include "nas5g_network.h"
+
+using namespace std;
+#define M5GS_REGISTRATION_RESULT_MAXIMUM_LENGTH 1  // TODO should be in nas5g
+#define INVALID_IMSI64 (imsi64_t) 0
+#define INVALID_AMF_UE_NGAP_ID 0x0
+
+namespace magma5g {
+extern ue_m5gmm_context_s
+    ue_m5gmm_global_context;  // TODO  global var to temporarily store
+amf_sap_c amf_sap_reg;
+nas_proc nas_procedure_reg;
+amf_as_data_t amf_data_sec;
+m5g_authentication m5g_auth;
+identification amf_proc;
+nas_network nas_nw;
+nas_amf_smc_proc_t smc_proc;
```

Comment:
what is the reason why all these are dangling as global variables? if you are only using the methods from these classes without any object data, simply define these functions or use class scope, e.g., `amf_sap_c::send(...)`.

Reply:
resolved